pkg/etcd/etcd.go

```diff
@@ -1162,46 +1209,67 @@ func (e *ETCD) trackLearnerProgress(ctx context.Context, progress *learnerProgre
 	var newCondition v1.NodeCondition
-	if promoted {
+	switch memberStatus {
+	case learnerStatus:
+		newCondition = v1.NodeCondition{
+			Type:    etcdStatusType,
+			Status:  "False",
+			Reason:  "MemberIsLearner",
+			Message: "Node has not been promoted to voting member of the etcd cluster",
+		}
+	case voterStatus:
 		newCondition = v1.NodeCondition{
 			Type:    etcdStatusType,
 			Status:  "True",
 			Reason:  "MemberNotLearner",
 			Message: "Node is a voting member of the etcd cluster",
 		}
-	} else {
+	case unhealthyStatus:
 		newCondition = v1.NodeCondition{
 			Type:    etcdStatusType,
 			Status:  "False",
-			Reason:  "MemberIsLearner",
-			Message: "Node has not been promoted to voting member of the etcd cluster",
+			Reason:  "Unhealthy",
+			Message: "Node is unhealthy",
+		}
+	case unjoinedStatus:
+		newCondition = v1.NodeCondition{
+			Type:    etcdStatusType,
+			Status:  "False",
+			Reason:  "NotAMember",
+			Message: "Node is not a member of the etcd cluster",
 		}
 	}
 
-	updatedNode := *node
-	if find, condition := nodeUtil.GetNodeCondition(&updatedNode.Status, etcdStatusType); find >= 0 {
-		if condition.Status == newCondition.Status {
-			logrus.Debugf("Member %s is not changing etcd status condition", memberName)
+	if find, condition := nodeUtil.GetNodeCondition(&node.Status, etcdStatusType); find >= 0 {
+		if condition.Status == newCondition.Status && memberStatus != unjoinedStatus {
+			logrus.Debugf("Node %s is not changing etcd status condition", memberName)
 			condition.LastHeartbeatTime = metav1.Now()
-			return nodeHelper.PatchNodeStatus(client, types.NodeName(node.Name), node, &updatedNode)
+			return nodeHelper.SetNodeCondition(client, types.NodeName(node.Name), *condition)
 		}
 
-		logrus.Debugf("Member %s is changing etcd condition", memberName)
+		logrus.Debugf("Node %s is changing etcd status condition", memberName)
 		condition = &newCondition
 		condition.LastHeartbeatTime = metav1.Now()
 		condition.LastTransitionTime = metav1.Now()
-		return nodeHelper.PatchNodeStatus(client, types.NodeName(node.Name), node, &updatedNode)
+		return nodeHelper.SetNodeCondition(client, types.NodeName(node.Name), *condition)
 	}
 
-	logrus.Infof("Adding etcd member %s status condition", memberName)
+	logrus.Infof("Adding node %s etcd status condition", memberName)
 	newCondition.LastHeartbeatTime = metav1.Now()
 	newCondition.LastTransitionTime = metav1.Now()
-	updatedNode.Status.Conditions = append(updatedNode.Status.Conditions, newCondition)
-	return nodeHelper.PatchNodeStatus(client, types.NodeName(node.Name), node, &updatedNode)
+	return nodeHelper.SetNodeCondition(client, types.NodeName(node.Name), newCondition)
+}
+
+func (e *ETCD) getEtcdHealth() (bool, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultDialTimeout)
+	defer cancel()
+	_, err := e.client.Status(ctx, e.clientURL())
```

Comment:
Check the response here - in particular, check for a non-empty `Errors []string`: https://pkg.go.dev/go.etcd.io/etcd/api/v3/etcdserverpb#StatusResponse

I also don't see that you're using this yet?